src/util-profiling.c

```diff
@@ -685,6 +691,58 @@ void SCProfilingDumpPacketStats(void)
         }
     }
 
+    fprintf(fp, "\nLogger/output stats:\n");
+
+    total = 0;
+    for (m = 0; m < LOGGER_SIZE; m++) {
+        int p;
+        for (p = 0; p < 257; p++) {
+            SCProfilePacketData *pd = &packet_profile_log_data4[m][p];
+            total += pd->tot;
+            pd = &packet_profile_log_data6[m][p];
+            total += pd->tot;
+        }
+    }
+
+    fprintf(fp, "\n%-24s   %-6s   %-5s   %-12s   %-12s   %-12s   %-12s   %-12s\n",
+            "Logger", "IP ver", "Proto", "cnt", "min", "max", "avg", "tot");
+    fprintf(fp, "%-24s   %-6s   %-5s   %-12s   %-12s   %-12s   %-12s   %-12s\n",
+            "------------------------", "------", "-----", "----------", "------------", "------------", "-----------", "-----------");
+    for (m = 0; m < LOGGER_SIZE; m++) {
+        int p;
+        for (p = 0; p < 257; p++) {
+            SCProfilePacketData *pd = &packet_profile_log_data4[m][p];
+
+            if (pd->cnt == 0) {
+                continue;
+            }
+
+            FormatNumber(pd->tot, totalstr, sizeof(totalstr));
+            double percent = (long double)pd->tot /
+                (long double)total * 100;
+
+            fprintf(fp, "%-24s    IPv4     %3d  %12"PRIu64"     %12"PRIu64"   %12"PRIu64"  %12"PRIu64"  %12s  %-6.2f\n",
+                    PacketProfileLoggertIdToString(m), p, pd->cnt, pd->min, pd->max, (uint64_t)(pd->tot / pd->cnt), totalstr, percent);
+        }
+    }
+    for (m = 0; m < LOGGER_SIZE; m++) {
+        int p;
+        for (p = 0; p < 257; p++) {
+            SCProfilePacketData *pd = &packet_profile_log_data6[m][p];
+
+            if (pd->cnt == 0) {
+                continue;
+            }
+
+            FormatNumber(pd->tot, totalstr, sizeof(totalstr));
+            double percent = (long double)pd->tot /
+                (long double)total * 100;
+
+            fprintf(fp, "%-24s    IPv6     %3d  %12"PRIu64"     %12"PRIu64"   %12"PRIu64"  %12"PRIu64"  %12s  %-6.2f\n",
+                    PacketProfileLoggertIdToString(m), p, pd->cnt, pd->min, pd->max, (uint64_t)(pd->tot / pd->cnt), totalstr, percent);
```

Comment:
On 24-08-16 16:04, Jason Ish wrote:
> In src/util-profiling.c
> <https://github.com/inliniac/suricata/pull/2181#discussion_r76062128>:
> 
>> +    }
>> +    for (m = 0; m < LOGGER_SIZE; m++) {
>> +        int p;
>> +        for (p = 0; p < 257; p++) {
>> +            SCProfilePacketData *pd = &packet_profile_log_data6[m][p];
>> +
>> +            if (pd->cnt == 0) {
>> +                continue;
>> +            }
>> +
>> +            FormatNumber(pd->tot, totalstr, sizeof(totalstr));
>> +            double percent = (long double)pd->tot /
>> +                (long double)total * 100;
>> +
>> +            fprintf(fp, "%-24s    IPv6     %3d  %12"PRIu64"     %12"PRIu64"   %12"PRIu64"  %12"PRIu64"  %12s  %-6.2f\n",
>> +                    PacketProfileLoggertIdToString(m), p, pd->cnt, pd->min, pd->max, (uint64_t)(pd->tot / pd->cnt), totalstr, percent);
> 
> Yes, but only because the output system is deregistered before the
> profiling output runs, so I lose my registration table to lookup the
> name by ID.
> 
> I suppose we could have a table mapping IDs to names as well,
> centralizing the creation of all the names that could be access by ID?

Let's safe it for later then.